.github/check_encoding.py

```
@@ -1,29 +1,75 @@
-        print("The following files are not in Windows-1252 encoding:")
-        for file in failed_files:
-            print(f"  - {file}")
+        file_name = os.path.basename(file_path)
+        results[file_path] = {
+            "encoding": is_windows1252_encoded(file_path),
+            "naming": is_valid_filename(file_name),
+            "folder": is_valid_folder(file_path)
+        }
+
+    failed = False
+    for file, checks in results.items():
```

Ruff (0.8.2)

Loop control variable `file` not used within loop body

(B007)